internal/backend/gs/gs.go

```diff
@@ -136,7 +136,14 @@ func Create(cfg Config, rt http.RoundTripper) (restic.Backend, error) {
 	ctx := context.Background()
 	exists, err := be.bucketExists(ctx, be.bucket)
 	if err != nil {
-		return nil, errors.Wrap(err, "service.Buckets.Get")
+		switch err.(type) {
```

Comment:
You almost got it! You could use a check similar to the following, this checks the type as well as the status code:

https://github.com/googleapis/google-cloud-go/blob/91b992b32bb3488a927338e7fd3203d9b43036eb/storage/bucket.go#L168-L170

Reply:
doh. forgot the status check. done.